public/app/features/variables/state/actions.ts

```diff
@@ -509,6 +513,10 @@ export const variableUpdated = (
 
     const variables = getVariables(getState());
     const g = createGraph(variables);
+    const affectedPanelIds = getAllAffectedPanelIdsForVariableChange(
+      variableInState.id,
+      getState().dashboard.getModel()
```

Comment:
is there a reason this is taking variables from the dashboard model instead of the variables from state?

Reply:
Do you mean within the getAllAffectedPanelIdsForVariableChange function?

Reply:
yes

Reply:
Good question, I wasn't thinking is the only answer I can come up with 🙈 . I'll of course need to use the variables in state instead. Good catch!